opensoundscape/annotations.py

```diff
@@ -1176,8 +1189,17 @@ def find_overlapping_idxs_in_clip_df(
         for row in clip_df.index
     ]
 
-    if min_label_overlap is not None:
+    # min_label_overlap is a required argument. If min_label_fraction is passed, it means
+    # we should keep annotations that have at least least min_label_fraction overlap
+    # EVEN if they have less than min_label_overlap seconds of overlap
+    if min_label_fraction is None:  # just use min_label_overlap
         clip_df = clip_df[clip_df["overlap"] >= min_label_overlap]
-    if min_label_fraction is not None:
-        clip_df = clip_df[clip_df["overlap_fraction"] >= min_label_fraction]
+    else:
+        # get all rows rows with at least min_label_fraction overlap
+        above_fraction = clip_df[clip_df["overlap_fraction"] >= min_label_fraction]
+        # we need to keep these AND any that have more than min_label_overlap
+        overlapping = clip_df[clip_df["overlap"] >= min_label_overlap]
+        # combine the two dataframes. but drop duplicates
+        clip_df = pd.concat([above_fraction, overlapping]).drop_duplicates()
```

Comment:
this will result in a weird ordering of the clips, rather than retaining the order. Can you use "or" logic on the two boolean masks `clip_df["overlap"] >= min_label_overlap` and `clip_df["overlap_fraction"] >= min_label_fraction` instead?

Reply:
Done